pkg/lightning/backend/external/writer.go

```diff
@@ -74,6 +74,38 @@ type rangePropertiesCollector struct {
 	propKeysDist uint64
 }
 
+// size: the file size after adding 'data'
+func (rc *rangePropertiesCollector) onNextEncodedData(data []byte, size uint64) {
+	keyLen := binary.BigEndian.Uint64(data)
+	key := data[2*lengthBytes : 2*lengthBytes+keyLen]
+
+	if len(rc.currProp.firstKey) == 0 {
+		rc.currProp.firstKey = key
+	}
+	rc.currProp.lastKey = key
+
+	rc.currProp.size += uint64(len(data) - 2*lengthBytes)
+	rc.currProp.keys++
+
+	if rc.currProp.size >= rc.propSizeDist ||
+		rc.currProp.keys >= rc.propKeysDist {
+		newProp := *rc.currProp
```

Comment:
I didn't open IDE to review. Just double check that the argument `data` should live longer than `rc` because `rc` refers to its content.

Reply:
this part is abstracted from `(s *KeyValueStore) addEncodedData`

but yes, before flush the content to s3, the data is remain unchanged